libkbfs/conflict_resolver.go

```diff
@@ -195,8 +206,18 @@ func (cr *ConflictResolver) Resolve(unmerged MetadataRevision,
 		return
 	}
 
+	ci := conflictInput{unmerged, merged}
+	func() {
+		cr.inputLock.Lock()
+		defer cr.inputLock.Unlock()
+		// Cancel any running CR before we return, so the caller can be
+		// confident any ongoing CR superseded by this new input will be
+		// canceled before it releases any locks it holds.
+		_ = cr.cancelExistingLocked(ci)
```

Comment:
I thought I would let it go through, just so we can get logging in the usual place when some input is rejected.  I guess I could pass a context into `Resolve` to make logging possible, but didn't feel like adding that biggish change as part of this PR. Will add a TODO.